core/domain/rights_manager.py

```diff
@@ -626,6 +626,10 @@ def _change_activity_status(
 
     if new_status != ACTIVITY_STATUS_PRIVATE:
```

Comment:
Can this part go after the _save_activity_rights? That way, if something goes wrong within _save_activity_rights, this doesn't run.